cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,798 @@
+
+    scores_overlooked = []
+    for iid, k in enumerate(pred_labels):
+        if pred_label_probs[iid] < high_probability_threshold:
+            continue
+
+        k_similarity = similarity_matrix[lab_labels == k, iid]
+        if len(k_similarity) == 0:  # if there is no annotated box
+            scores_overlooked.append(min_possible_similarity * (1 - pred_label_probs[iid]))
+        else:
+            closest_annotated_box = np.argmax(k_similarity)
+            scores_overlooked.append(k_similarity[closest_annotated_box])
+    return scores_overlooked
+
+
+def _compute_scores_badloc_for_image(
+    pred_labels,
+    pred_label_probs,
+    lab_labels,
+    similarity_matrix,
+    low_probability_threshold,
+    min_possible_similarity,
+):
+    """Function to compute the bad location score for each label in a single image."""
+    scores_badloc = []
+    for iid, k in enumerate(lab_labels):  # for every annotated box
+        k_similarity = similarity_matrix[iid, pred_labels == k]
+        k_pred = pred_label_probs[pred_labels == k]
+
+        if len(k_pred) == 0:  # there are no predicted boxes of class k
+            scores_badloc.append(min_possible_similarity)
+            continue
+
+        idx_at_least_low_probability_threshold = k_pred > low_probability_threshold
+        k_similarity = k_similarity[idx_at_least_low_probability_threshold]
+        k_pred = k_pred[idx_at_least_low_probability_threshold]
+        assert len(k_pred) == len(k_similarity)
+        if len(k_pred) == 0:
+            scores_badloc.append(min_possible_similarity)
+        else:
+            scores_badloc.append(np.max(k_similarity))
+    return scores_badloc
+
+
+def _compute_scores_swap_for_image(
+    pred_labels,
+    pred_label_probs,
+    lab_labels,
+    similarity_matrix,
+    high_probability_threshold,
+    min_possible_similarity,
+):
+    """Function to compute the swap score for each label in a single image."""
+    scores_swap = []
+    for iid, k in enumerate(lab_labels):
+        not_k_idx = pred_labels != k
+
+        if len(not_k_idx) == 0:
+            scores_swap.append(1.0)
+            continue
+
+        not_k_similarity = similarity_matrix[iid, not_k_idx]
+        not_k_pred = pred_label_probs[not_k_idx]
+
+        idx_at_least_high_probability_threshold = not_k_pred > high_probability_threshold
+        if len(idx_at_least_high_probability_threshold) == 0:
+            scores_swap.append(1.0)
+            continue
+
+        not_k_similarity = not_k_similarity[idx_at_least_high_probability_threshold]
+        if len(not_k_similarity) == 0:  # if there is no annotated box
+            scores_swap.append(1.0)
+        else:
+            closest_predicted_box = np.argmax(not_k_similarity)
+            score = np.max([min_possible_similarity, 1 - not_k_similarity[closest_predicted_box]])
+            scores_swap.append(score)
+    return scores_swap
+
+
+def _compute_subtype_lqs(
```

Comment:
re naming try to follow the package conventions, eg from cleanlab.rank.

So consider calling this: `_get_subtype_label_quality_scores`